apollo-router/src/configuration/shared/mod.rs

```diff
@@ -3,8 +3,25 @@ use serde::Deserialize;
 
 use crate::plugins::traffic_shaping::Http2Config;
 
-#[derive(Debug, Clone, Default, Deserialize, JsonSchema)]
+#[derive(PartialEq, Debug, Clone, Default, Deserialize, JsonSchema, buildstructor::Builder)]
 #[serde(deny_unknown_fields, default)]
 pub(crate) struct Client {
     pub(crate) experimental_http2: Option<Http2Config>,
+    pub(crate) dns_resolution_strategy: Option<DnsResolutionStrategy>,
+}
```

Comment:
I tried to use this share structure as much as possible to unify the behavior between the coprocessor's and the subgraph's config.
Maybe in Router2, we can unify it even thither and change subgraph config to look like:
```
traffic_shaping:
  all:
    client:
      dns_resolution_strategy: ipv4_then_ipv6
```